wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Core State & Navigation Functionality with Tags screen integration

Fixes #11634

## Findings
This PR hopes to add the tags screen functionality to the Prepublishing bottom sheet while allowing proper state management and navigation.

## Solution
### Current architecture 
* The `TagsFragment` is being reused that was extracted in https://github.com/wordpress-mobile/WordPress-Android/pull/11644 As previously, discussed this was done for code & functionality reusability. 
* There is a `PrepublishingActionsViewModel` that creates the main screen with the relevant actions such as Publish, Tags, Visibility etc. Currently, only the Tags action performs navigation. 
* The parent `PrepublishingViewModel` is where the core state will be stored. It contains logic to persist the selected values for the different options using the `EditPostRepository`. 

## Testing
1. Create a new post. 
2. Press publish. 
3. Select the Tags field. 
4. See the tags screen. 
5. Add tags and press back so they can be saved.
6. You can also collapse the whole bottom sheet using the close button.
7. Test different configuration changes such as rotations, kill the activity etc. 
8. Go to the Post Settings to ensure that the tags actually persist when changes are made. 
N.B I saw a bug where going back to the home screen and back to the tags screen when tags are set actually causes multiple commas to be added. 

-----------------------------
Unit tests for the `PrepublishingViewModel` will be written once this PR is merged. It wasn't added since the PR grew a bit in size. 

Android Design (Updated today) | Current 
--------|-------
  <img width="358" alt="Screen Shot 2020-04-13 at 10 00 57 PM" src="https://user-images.githubusercontent.com/1509205/79182070-80c22500-7dd3-11ea-8b6c-8d8c48e7bfc1.png">   |  <img src="https://user-images.githubusercontent.com/1509205/79182091-90da0480-7dd3-11ea-964d-e3a48b88338f.png" width="320">

Android Design (Updated today) | Current 
--------|-------
  <img width="358" alt="Screen Shot 2020-04-13 at 10 00 49 PM" src="https://user-images.githubusercontent.com/1509205/79182067-7f90f800-7dd3-11ea-9e8b-821cd6a2d343.png">      | <img src="https://user-images.githubusercontent.com/1509205/79300431-4a0c0d80-7eac-11ea-8db1-d08eccc33b46.png" width="320">
